engine/consensus/compliance/core_test.go

```diff
@@ -332,40 +332,156 @@ func (cs *CoreSuite) TestOnBlockProposalSkipProposalThreshold() {
 
 	// block should be dropped - not added to state or cache
 	cs.state.AssertNotCalled(cs.T(), "Extend", mock.Anything)
+	cs.validator.AssertNotCalled(cs.T(), "ValidateProposal", mock.Anything)
 	cs.pending.AssertNotCalled(cs.T(), "Add", originID, mock.Anything)
 }
 
-func (cs *CoreSuite) TestOnBlockProposalInvalidExtension() {
+// TestOnBlockProposal_InvalidProposal tests that a proposal which fails HotStuff validation.
```

Comment:
similar naming suggestion as for collector compliance tests (see [my comment here](https://github.com/onflow/flow-go/pull/3294#issuecomment-1278325803))

Reply:
See https://github.com/onflow/flow-go/pull/3294#issuecomment-1282354892